specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json

```diff
@@ -0,0 +1,2146 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/reservationOrderIdParameter"
+          },
+          {
+            "name": "grain",
+            "description": "Can be daily or monthly",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "daily",
+              "monthly"
+            ],
+            "x-ms-enum": {
+              "name": "datagrain",
+              "modelAsString": true,
+              "values": [
+                {
+                  "value": "daily",
+                  "description": "Daily grain of data",
+                  "name": "DailyGrain"
+                },
+                {
+                  "value": "monthly",
+                  "description": "Monthly grain of data",
+                  "name": "MonthlyGrain"
+                }
+              ]
+            }
+          },
+          {
+            "name": "$filter",
+            "description": "Required only for daily grain. The properties/UsageDate for start date and end date. The filter supports 'le' and  'ge'",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationSummariesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Capacity/reservationorders/{reservationOrderId}/reservations/{reservationId}/providers/Microsoft.Consumption/reservationSummaries": {
+      "get": {
+        "tags": [
+          "ReservedInstances"
+        ],
+        "operationId": "ReservationsSummaries_ListByReservationOrderAndReservation",
+        "description": "Lists the reservations summaries for daily or monthly grain.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/consumption/"
+        },
+        "x-ms-examples": {
+          "ReservationSummariesDailyWithReservationId": {
+            "$ref": "./examples/ReservationSummariesDailyWithReservationId.json"
+          },
+          "ReservationSummariesMonthlyWithReservationId": {
+            "$ref": "./examples/ReservationSummariesMonthlyWithReservationId.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/reservationOrderIdParameter"
+          },
+          {
+            "$ref": "#/parameters/reservationIdParameter"
```

Comment:
Looks like this is a  duplicate of this parameter: https://github.com/Azure/azure-rest-api-specs/pull/2625/files#diff-dca9b11ab859f276378213344e091b34R342

Moving the definition into the "defintions" section and referencing it both places could improve the maintainability of this document moving forward.

Reply:
reservationOrderIdParameter and reservationIdParameter are two different parameters which are going to be part of the URI so we have defined them as Parameters. 

Please let me now if I missed the point.

Reply:
Ah sorry, it looks like my comment is based on the same line as it was before or something. I'm speaking to "grain" and "datagrain" enums.
 
Unless there's a subtle difference that I'm just overlooking, It seems that the following definition appears multiple times in this PR:

``` JSON
          {
            "name": "grain",
            "description": "Can be daily or monthly",
            "in": "query",
            "required": true,
            "type": "string",
            "enum": [
              "daily",
              "monthly"
            ],
            "x-ms-enum": {
              "name": "datagrain",
              "modelAsString": true,
              "values": [
                {
                  "value": "daily",
                  "description": "Daily grain of data",
                  "name": "DailyGrain"
                },
                {
                  "value": "monthly",
                  "description": "Monthly grain of data",
                  "name": "MonthlyGrain"
                }
              ]
            }
          }
```

Reply:
Thank you! for sharing more details. I have made the necessary changes and updated the PR.